segfault when using step ssh login's --add-user with Google provisioner

### Subject of the issue

Using `--add-user` with `step ssh login` and a Google provisioner causes a segfault in cli 0.14.2. The normal ssh certificate is issued and visible in `step ssh list`, but the provisioning certificate is not issued. Not sure if this affects all OIDC provisioners or just Google.

`step ssh login` works fine against the Google provisioner without `--add-user`. `--add-user` can be used successfully against the default JWK provisioner (both normal and provisioning certs are issued).

Google OIDC provisioner was registered to CA with the `--ssh` flag:

```
step ca provisioner add google --type oidc --ssh \
    --client-id ... \
    --client-secret ... \
    --configuration-endpoint https://accounts.google.com/.well-known/openid-configuration
```

Google OIDC provisioner config:

```
   {
      "type": "OIDC",
      "name": "google",
      "clientID": "...",
      "clientSecret": "...",
      "configurationEndpoint": "https://accounts.google.com/.well-known/openid-configuration",
      "claims": {
         "enableSSHCA": true
      }
   }
```


### Your environment
* OS - Debian 10 (in Docker)
* Version - cli 0.14.2, certificates 0.14.3

### Steps to reproduce

* set up step-ca with Google provisioner
* bootstrap client to step-ca
* on client, request a provisioning ssh certificate with `--add-user` against the Google provisioner

```
# step ssh login me@mydomain.com --provisioner google --add-user                                                                                                                                           
```

### Expected behaviour
Two ssh certificates are issued (normal cert and provisioning cert).

### Actual behaviour
cli segfaults. The normal cert is issued and visible in `step ssh list`.

### Additional context

```
# STEPDEBUG=1 step ssh login me@mydomain.com --provisioner google --add-user                                                                                                                                           
✔ Provisioner: google (OIDC) [client: client-id...]

Cannot open a web browser on your platform.                                                                                                                                                                                                   
                                                                                                                                                                                                                                              
Open a local web browser and visit:                                                                                                                                                                                                           
                                                                                                                                                                                                                                              
https://accounts.google.com/o/oauth2/v2/auth?client_id=...                              
                                                                                                                                                                                                                                              
✔ CA: https://step-ca:8443                                                                                                                                                                                                                    
✔ SSH Agent: yes                                                                                                                                                                                                                              
Smallstep CLI/0.14.2 (linux/amd64)                                                                      
Release Date: 2020-04-10 03:41 UTC                                                     
                                                                                 
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference 
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xce28d7]             


──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                                                                                                                                                    [297/1991]
goroutine 1 [running]:                                            
main.panicHandler()                                                       
        /home/travis/gopath/src/github.com/smallstep/cli/cmd/step/main.go:117 +0x360
panic(0xddb540, 0x17eba30)                                                     
        /home/travis/.gimme/versions/go1.14.2.linux.amd64/src/runtime/panic.go:969 +0x166
github.com/smallstep/cli/command/ssh.loginAction(0xc000150420, 0x0, 0x0)  
        /home/travis/gopath/src/github.com/smallstep/cli/command/ssh/login.go:225 +0x9b7
github.com/smallstep/cli/command.ActionFunc.func1(0xc000150420, 0x0, 0x0)
        /home/travis/gopath/src/github.com/smallstep/cli/command/command.go:48 +0x3f
github.com/urfave/cli.HandleAction(0xdc7380, 0xc000342fe0, 0xc000150420, 0xc000150420, 0x0)
        /home/travis/gopath/pkg/mod/github.com/urfave/cli@v1.22.2/app.go:521 +0xc1
github.com/urfave/cli.Command.Run(0xf09c73, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, 0xf612a0, 0x34, 0xf7dc2c, ...)
        /home/travis/gopath/pkg/mod/github.com/urfave/cli@v1.22.2/command.go:174 +0x56e
github.com/urfave/cli.(*App).RunAsSubcommand(0xc0000141c0, 0xc000150000, 0x0, 0x0)
        /home/travis/gopath/pkg/mod/github.com/urfave/cli@v1.22.2/app.go:404 +0x909
github.com/urfave/cli.Command.startApp(0xf075b2, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0xf4c229, 0x22, 0xf6bb42, ...)
        /home/travis/gopath/pkg/mod/github.com/urfave/cli@v1.22.2/command.go:373 +0x845
github.com/urfave/cli.Command.Run(0xf075b2, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0xf4c229, 0x22, 0xf6bb42, ...)
        /home/travis/gopath/pkg/mod/github.com/urfave/cli@v1.22.2/command.go:102 +0xa1f
github.com/urfave/cli.(*App).Run(0xc000014000, 0xc000130000, 0x7, 0x7, 0x0, 0x0)
        /home/travis/gopath/pkg/mod/github.com/urfave/cli@v1.22.2/app.go:276 +0x7bb
main.main()
        /home/travis/gopath/src/github.com/smallstep/cli/cmd/step/main.go:93 +0x3f3
```

--- a/command/ssh/certificate.go
+++ b/command/ssh/certificate.go
@@ -446,7 +446,7 @@ func certificateAction(ctx *cli.Context) error {
 	}
 
 	// Write Add User keys and certs
-	if isAddUser {
+	if isAddUser && resp.AddUserCertificate != nil {
 		id := provisioner.SanitizeSSHUserPrincipal(subject) + "-provisioner"
 		if _, err := pemutil.Serialize(auPriv, pemutil.WithOpenSSH(true), pemutil.ToFile(baseName+"-provisioner", 0600)); err != nil {
 			return err
@@ -486,10 +486,10 @@ func certificateAction(ctx *cli.Context) error {
 		}
 	}
 
-	if isAddUser {
-		ui.PrintSelected("Provisioner Private Key", baseName+"-provisioner")
-		ui.PrintSelected("Provisioner Public Key", baseName+"-provisioner.pub")
-		ui.PrintSelected("Provisioner Certificate", baseName+"-provisioner-cert.pub")
+	if isAddUser && resp.AddUserCertificate != nil {
+		ui.PrintSelected("Add User Private Key", baseName+"-provisioner")
+		ui.PrintSelected("Add User Public Key", baseName+"-provisioner.pub")
+		ui.PrintSelected("Add User Certificate", baseName+"-provisioner-cert.pub")
 	}
 
 	return nil

--- a/command/ssh/login.go
+++ b/command/ssh/login.go
@@ -223,7 +223,7 @@ func loginAction(ctx *cli.Context) error {
 	}
 	if isAddUser {
 		if resp.AddUserCertificate == nil {
-			ui.Printf(`{{ "%s" | red }} {{ "Add User:" | bold }} failed to create a provisioner certificate.`+"\n", ui.IconBad)
+			ui.Printf(`{{ "%s" | red }} {{ "Add User:" | bold }} failed to create a provisioner certificate`+"\n", ui.IconBad)
 		} else if err := agent.AddCertificate(subject, resp.AddUserCertificate.Certificate, auPriv); err != nil {
 			ui.Printf(`{{ "%s" | red }} {{ "Add User:" | bold }} %v`+"\n", ui.IconBad, err)
 		} else {

--- a/command/ssh/login.go
+++ b/command/ssh/login.go
@@ -222,10 +222,12 @@ func loginAction(ctx *cli.Context) error {
 		ui.PrintSelected("SSH Agent", "yes")
 	}
 	if isAddUser {
-		if err := agent.AddCertificate(subject, resp.AddUserCertificate.Certificate, auPriv); err != nil {
-			ui.Printf(`{{ "%s" | red }} {{ "SSH Agent:" | bold }} %v`+"\n", ui.IconBad, err)
+		if resp.AddUserCertificate == nil {
+			ui.Printf(`{{ "%s" | red }} {{ "Add User:" | bold }} failed to create a provisioner certificate.`+"\n", ui.IconBad)
+		} else if err := agent.AddCertificate(subject, resp.AddUserCertificate.Certificate, auPriv); err != nil {
+			ui.Printf(`{{ "%s" | red }} {{ "Add User:" | bold }} %v`+"\n", ui.IconBad, err)
 		} else {
-			ui.PrintSelected("SSH Agent", "yes")
+			ui.PrintSelected("Add User", "yes")
 		}
 	}
 

--- a/command/ssh/login.go
+++ b/command/ssh/login.go
@@ -223,11 +223,11 @@ func loginAction(ctx *cli.Context) error {
 	}
 	if isAddUser {
 		if resp.AddUserCertificate == nil {
-			ui.Printf(`{{ "%s" | red }} {{ "Add User:" | bold }} failed to create a provisioner certificate`+"\n", ui.IconBad)
+			ui.Printf(`{{ "%s" | red }} {{ "Add User Certificate:" | bold }} failed to create a provisioner certificate`+"\n", ui.IconBad)
 		} else if err := agent.AddCertificate(subject, resp.AddUserCertificate.Certificate, auPriv); err != nil {
-			ui.Printf(`{{ "%s" | red }} {{ "Add User:" | bold }} %v`+"\n", ui.IconBad, err)
+			ui.Printf(`{{ "%s" | red }} {{ "Add User Certificate:" | bold }} %v`+"\n", ui.IconBad, err)
 		} else {
-			ui.PrintSelected("Add User", "yes")
+			ui.PrintSelected("Add User Certificate", "yes")
 		}
 	}
 

